syntax = "proto3";
package tr451_vomci_function_sbi_message;

message VomciFunctionHello {
  string            vomci_name = 1; //vOMCI instance name
}

message OltHello {
  string            olt_name = 1;   //pOLT name
}

message HelloVomciRequest {
  oneof local_endpoint_hello {
    VomciFunctionHello vomci_function = 1;
    OltHello olt = 2;
  }
}

message HelloVomciResponse {
  oneof remote_endpoint_hello {
    VomciFunctionHello vomci_function = 1;
    OltHello olt = 2;
  }
}

message OmciPacket {
  string   onu_id         = 1; //The TC layer ONU-ID identifier of the TR-383 vANI
  string   chnl_term_name = 2; //The reference identifier to the channel termination of the TR-383 vANI
  bytes    payload        = 3; //OMCI message payload
}
